Logger/BaseLogger.cs

```diff
@@ -3,5 +3,7 @@
 public abstract class BaseLogger
 {
     public abstract void Log(LogLevel logLevel, string message);
+    public string? ClassName
+        { get; set; }
```

Comment:
```suggestion
    public string ClassName { get; set; }
```
Getter and setter can be on same line preferably in this case.
Why is this nullable?